### PR TITLE
LPD-20815 Add support to use null in filter for Object Proxy Salesforce

### PR DESCRIPTION
**Pull request by jairmedeiros**
Related issue: https://liferay.atlassian.net/browse/LPD-20813

### CODE REVIEW

File: modules/apps/object/object-storage-salesforce-test/src/testIntegration/java/com/liferay/object/storage/salesforce/internal/rest/manager/v1_0/test/SalesforceObjectEntryManagerImplTest.java [hunk 1]
```diff
@@ -402,6 +403,14 @@ public void testGetObjectEntries() throws Exception {
 			).build(),
 			objectEntry2, objectEntry3, objectEntry4);
 
+		testGetObjectEntries(
+			HashMapBuilder.put(
+				"filter",
+				filterString.concat(
+					buildEqualsExpressionFilterString("title", null))
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-storage-salesforce-test/src/testIntegration/java/com/liferay/object/storage/salesforce/internal/rest/manager/v1_0/test/SalesforceObjectEntryManagerImplTest.java:410]: Hey @jairmedeiros is this causing a NPE at https://github.com/liferay/liferay-portal/blob/08475a044218ee312bc6452f6348f6d686cf8f6c/modules/apps/object/object-rest-test-util/src/main/java/com/liferay/object/rest/test/util/BaseObjectEntryManagerImplTestCase.java#L97?

File: modules/apps/object/object-storage-salesforce/src/main/java/com/liferay/object/storage/salesforce/internal/odata/filter/expression/SOSQLExpressionVisitorImpl.java [hunk 2]
```diff
@@ -152,6 +153,10 @@ public Object visitLiteralExpression(LiteralExpression literalExpression)
 				StringPool.APOSTROPHE);
 		}
 
+		if (Objects.equals(null, literalExpression.getType())) {
+			return null;
+		}
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-storage-salesforce/src/main/java/com/liferay/object/storage/salesforce/internal/odata/filter/expression/SOSQLExpressionVisitorImpl.java:158]: Hey @jairmedeiros we can move this conditional to be checked first this way we won't need to verify if the liferay Expression.getType is not null in the conditional above.

Thank you!

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]
Hey @jairmedeiros see my comments!

Thanks!

**Comment by jairmedeiros** [comment]
@MarinhoFeliphe, done!

File: modules/apps/object/object-storage-salesforce-test/src/testIntegration/java/com/liferay/object/storage/salesforce/internal/rest/manager/v1_0/test/SalesforceObjectEntryManagerImplTest.java [hunk 3]
```diff
@@ -402,6 +403,12 @@ public void testGetObjectEntries() throws Exception {
 			).build(),
 			objectEntry2, objectEntry3, objectEntry4);
 
+		testGetObjectEntries(
+			HashMapBuilder.put(
+				"filter", "title eq null"
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-storage-salesforce-test/src/testIntegration/java/com/liferay/object/storage/salesforce/internal/rest/manager/v1_0/test/SalesforceObjectEntryManagerImplTest.java:408]: Hey @jairmedeiros use `buildEqualsExpressionFilterString` here, I believe you have to adapt the method [getValue](https://github.com/liferay/liferay-portal/blob/08475a044218ee312bc6452f6348f6d686cf8f6c/modules/apps/object/object-rest-test-util/src/main/java/com/liferay/object/rest/test/util/BaseObjectEntryManagerImplTestCase.java#L118) to handle the null use case.

Thank you"

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]
Hey @jairmedeiros see my comments,

Thank you!

**Review by MarinhoFeliphe — APPROVED** [review]
LGTM!! :tada:

**Comment by MarinhoFeliphe** [comment]
ci:forward

**Comment by MarinhoFeliphe** [comment]
Resent here: https://github.com/brianchandotcom/liferay-portal/pull/148984

**Pull request closed without merging**